Maxq

anhu opened the pull request.

anhu committed:
Applied the patch we got via sharepoint.

There were two rejections. I couldn't figure out why they were rejected.
Did the merge by hand (copy+paste).  Otherwise, no other changes yet.
Next commit will add the .rej file.

anhu committed:
The .rej file for the hand integrated code I added.

anhu committed:
new files added by the patch.

anhu committed:
Fix a build error when maxq is not enabled.

anhu committed:
Remove the .rej file

anhu committed:
Get rid of all FORCE guarded code

anhu committed:
Move some defines to c file

anhu committed:
Error out if you enable ASN template and MAXQ.

anhu committed:
Properly initialize/override rng devId for maxq

anhu committed:
Whitespace and formatting changes

anhu committed:
Always use maxq10xx PK Callbacks when client in TLS 1.3.

anhu committed:
Make strstr_with_size use XSTRLEN and XSTRNCMP

anhu committed:
Get rid of redunant if statement

anhu committed:
whitespace

anhu committed:
Get rid of variable length arrays

anhu committed:
Properly check MXQ_* API return codes.

anhu committed:
Return codes for wc_MAXQ10XX_EccSetKey() and wc_MAXQ10XX_EccSetKey()

anhu committed:
One less call to wolfSSL_CryptHwMutexUnLock

anhu committed:
Ensure wolfSSL_CryptHwMutexLock() return is always checked

anhu committed:
cert len can be 1024, keep ssl->buffers.sig.buffer around,

....remove feature not compiled as suggested via email

anhu committed:
maxq10xx_random() does not need to be exposed.

mbayraker reviewed (changes requested):
I think "One less call to wolfSSL_CryptHwMutexUnLock" commit should be reverted.

mbayraker commented on wolfcrypt/src/port/maxim/maxq10xx.c:
```diff
@@ -609,15 +609,14 @@ void wc_MAXQ10XX_AesFree(Aes* aes)
     if (rc == 0) {
         free_aes_key_id(aes->maxq_ctx.key_obj_id);
         aes->maxq_ctx.key_obj_id = 0;
-        wolfSSL_CryptHwMutexUnLock();
```
This may cause problems during the allocation. Unlock should be called after free_aes_key_id(). Previous version is much safer.

anhu replied:
Excellent!!  Thank you for letting me know. I will make an adjustment.

mbayraker commented on wolfcrypt/src/port/maxim/maxq10xx.c:
```diff
@@ -133,112 +123,6 @@ int device_hs_key_type;
 #endif
 
 #ifdef WOLFSSL_MAXQ10XX_CRYPTO
-/*
```
Not a dead code.

anhu replied:
Inquired about this via email.

mbayraker reviewed (changes requested):
The whole commit about the "wolfSSL_MAXQ10XX_CryptoDevCb()" is not good and should be reverted.

anhu committed:
Fix as suggested by mbayraker

anhu committed:
Simplify the rng calls in the callback

anhu committed:
Set MAXQ_DEVICE_ID when we setup callbacks.

anhu committed:
Bring the HMAC integration into the callback.

anhu committed:
missing else

anhu committed:
Function renaming for consistency.

maxq10xx_SetPssSignature --> maxq10xx_SetRsaPssSignature
wc_MAXQ10XX_HKDF_Expand --> maxq10xx_HkdfExpand

anhu committed:
Get ready to make HKDF expand a PK Callback

anhu committed:
Move the DH Keygen callback to the TLS layer.

anhu commented:
> The whole commit about the "wolfSSL_MAXQ10XX_CryptoDevCb()" is not good and should be reverted.
Agreed.  Reverted.

alperenguclu commented on src/internal.c:
```diff
@@ -5854,19 +5849,18 @@ int DhGenKeyPair(WOLFSSL* ssl, DhKey* dhKey,
     WOLFSSL_ENTER("DhGenKeyPair");
 
+#if defined(HAVE_PK_CALLBACKS) && defined(WOLFSSL_MAXQ108x)
+    ret = maxq10xx_DhGenerateKeyPair(dhKey, ssl->rng, priv, privSz, pub, pubSz);
```
Looks good.

anhu committed:
Add the new DhGenerateKeyPair callback.

anhu committed:
Add a new HstypeAndKeylen callback.

anhu committed:
Add a new HstypeAndSiglen callback.

This also introduces new flag to let maxq know about TLS 1.3.
That's because this one is specifically for TLS 1.2.

anhu committed:
Introduce the new ReadCertDerCb callback.

anhu committed:
Add comment about test key

anhu committed:
These changes ensure that things work on the MAXQ1065!

anhu committed:
Get rid of maxq10xx_SetTls13Side()

anhu committed:
unused forSide

anhu committed:
Pragma Uninitialized

anhu committed:
Almost all MAXQ integration converted into callback

anhu committed:
Get rid of SignCertCb. Redundancy.

anhu committed:
Get rid of PerformClientFinCb. Redundancy.

Note that this break backwards compatibility. Permission was given by dgarske.

anhu committed:
Get rid of PerformClientKexCb. Redundancy.

anhu committed:
Get rid of Tls13RecordProcessingCb. Redundancy

anhu committed:
Get rid of HKDFExpandKeyLabelCb. Redundancy.

anhu committed:
Get rid of HstypeAndSiglenCb; introduces the need for pub key as priv key on command line.

anhu committed:
Get rid of HstypeAndKeylenCb

anhu commented:
Further review will continue at https://github.com/anhu/wolfssl/pull/9
